lib/calculator/parser.rb

```diff
@@ -63,7 +63,21 @@ def shift_tokens
 
     # Entry point for the parser, parses an <em>expression</em>
     def parse_expression
-      # TODO implement me
+      ExpressionNode.new(parse_term,parse_expression_prime)
     end
+
+    def parse_base
+      case @tokens.first
+        when IntegerToken
+          BaseNode.new(shift_tokens)
+        when DecimalToken
+          BaseNode.new(shift_tokens)
+        when LeftParenthesisToken
+          BaseNode.new(shift_tokens, parse_expression,shift_tokens)
```

Comment:
The tricky part here that you are possibly missing is that you are not checking to see if the next token after parse_expression is a right parenthesis.  As such, it's possible that some syntax errors may go unnoticed.